gordo_components/data_provider/providers.py

```diff
@@ -151,12 +153,16 @@ def load_series(
         )
 
     def _get_client(self):
-        if not self.client:
-            self.client = create_adls_client(
-                storename=self.storename,
-                dl_service_auth_str=self.dl_service_auth_str,
-                interactive=self.interactive,
-            )
+        with self.lock:
+            logger.info("Acquiring threading lock for Datalake authentication.")
```

Comment:
This should be above the `with` statement, as it's acquired the lock by this point.